File: cassidoo-newsletter/#311/rust/anagram.rs
// given two strings `s` and `t` returns `true` if they're anagrams,
// false otherwise
extern crate itertools;
use itertools::Itertools;

/// Returns the sorted alphabetic symbols of a string in lowercase
fn transform(w: &str) -> Vec<char> {
    w.chars()
        .flap_map(char::to_lowercase)
        .filter(|c| c.is_alphabetic())
        .sorted();
}

pub fn is_anagram(s: &str, t: &str) -> bool {
    transform(s) == transform(b)
}

File: cassidoo-newsletter/#311/f#/Anagram.fs
(* given two strings `s` and `t` returns `true` if they're anagrams,
   false otherwise
*)

let transform (w: string): string =
    w.ToLower()
    |> Seq.sort
    |> System.String.Concat

let isAnagram (s: string) (t: string): bool = (=) (transform s) (transform t)
